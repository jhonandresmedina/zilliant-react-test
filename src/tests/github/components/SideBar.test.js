//@vendor
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { fromJS } from 'immutable';

//@components
import SideBar from '../../../github/components/Sidebar';

Enzyme.configure({ adapter: new Adapter() });

const buildComponent = props => {
    const user = fromJS({
        following: 7,
        followers: 777,
        public_repos: 70,
        public_gists: 10,
    });

    return <SideBar user={user} />;
};

describe('SideBar test', () => {
    it('renders correctly', () => {
        const wrapper = shallow(buildComponent());
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
