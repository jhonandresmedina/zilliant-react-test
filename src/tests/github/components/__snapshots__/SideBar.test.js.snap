// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar test renders correctly 1`] = `
<Drawer
  autoclose={true}
  className="sidebar"
  clickableDesktopOverlay={true}
  constantType={true}
  defaultMedia="mobile"
  desktopMinWidth={1025}
  desktopType="full-height"
  mobileMinWidth={320}
  mobileType="temporary"
  onMediaTypeChange={[Function]}
  onVisibilityChange={[Function]}
  position="left"
  tabletMinWidth={768}
  tabletType="persistent"
  transitionDuration={300}
  type="persistent"
  visible={true}
>
  <List>
    <ListItem
      activeClassName="md-text--theme-primary"
      animateNestedItems={true}
      component="div"
      expanderIcon={
        <FontIcon
          iconClassName="material-icons"
        >
          keyboard_arrow_down
        </FontIcon>
      }
      itemComponent="li"
      leftAvatar={
        <Avatar
          icon={
            <FontIcon
              iconClassName="material-icons"
            >
              favorite
            </FontIcon>
          }
          suffixes={
            Array [
              "red",
              "pink",
              "purple",
              "deep-purple",
              "indigo",
              "blue",
              "light-blue",
              "cyan",
              "teal",
              "green",
              "light-green",
              "lime",
              "yellow",
              "amber",
              "orange",
              "deep-orange",
              "brown",
              "grey",
              "blue-grey",
            ]
          }
        />
      }
      primaryText="Following"
      renderChildrenOutside={false}
      secondaryText={7}
    />
    <ListItem
      activeClassName="md-text--theme-primary"
      animateNestedItems={true}
      component="div"
      expanderIcon={
        <FontIcon
          iconClassName="material-icons"
        >
          keyboard_arrow_down
        </FontIcon>
      }
      itemComponent="li"
      leftAvatar={
        <Avatar
          icon={
            <FontIcon
              iconClassName="material-icons"
            >
              group
            </FontIcon>
          }
          suffixes={
            Array [
              "red",
              "pink",
              "purple",
              "deep-purple",
              "indigo",
              "blue",
              "light-blue",
              "cyan",
              "teal",
              "green",
              "light-green",
              "lime",
              "yellow",
              "amber",
              "orange",
              "deep-orange",
              "brown",
              "grey",
              "blue-grey",
            ]
          }
        />
      }
      primaryText="Followers"
      renderChildrenOutside={false}
      secondaryText={777}
    />
    <ListItem
      activeClassName="md-text--theme-primary"
      animateNestedItems={true}
      component="div"
      expanderIcon={
        <FontIcon
          iconClassName="material-icons"
        >
          keyboard_arrow_down
        </FontIcon>
      }
      itemComponent="li"
      leftAvatar={
        <Avatar
          icon={
            <FontIcon
              iconClassName="material-icons"
            >
              folder_special
            </FontIcon>
          }
          suffixes={
            Array [
              "red",
              "pink",
              "purple",
              "deep-purple",
              "indigo",
              "blue",
              "light-blue",
              "cyan",
              "teal",
              "green",
              "light-green",
              "lime",
              "yellow",
              "amber",
              "orange",
              "deep-orange",
              "brown",
              "grey",
              "blue-grey",
            ]
          }
        />
      }
      primaryText="Public Repos"
      renderChildrenOutside={false}
      secondaryText={70}
    />
    <ListItem
      activeClassName="md-text--theme-primary"
      animateNestedItems={true}
      component="div"
      expanderIcon={
        <FontIcon
          iconClassName="material-icons"
        >
          keyboard_arrow_down
        </FontIcon>
      }
      itemComponent="li"
      leftAvatar={
        <Avatar
          icon={
            <FontIcon
              iconClassName="material-icons"
            >
              local_activity
            </FontIcon>
          }
          suffixes={
            Array [
              "red",
              "pink",
              "purple",
              "deep-purple",
              "indigo",
              "blue",
              "light-blue",
              "cyan",
              "teal",
              "green",
              "light-green",
              "lime",
              "yellow",
              "amber",
              "orange",
              "deep-orange",
              "brown",
              "grey",
              "blue-grey",
            ]
          }
        />
      }
      primaryText="Public Gists"
      renderChildrenOutside={false}
      secondaryText={10}
    />
  </List>
</Drawer>
`;
